rypto/rsa: 512-bit keys are insecure : fix tests

```
% got                                                                                                                                                                                                   ~miekg/dns master
--- FAIL: TestNewPrivateKey (0.00s)
    parse_test.go:1269: crypto/rsa: 512-bit keys are insecure (see https://go.dev/pkg/crypto/rsa#hdr-Minimum_key_size)
--- FAIL: TestSIG0 (0.05s)
    sig0_test.go:33: failed to generate key for "RSASHA1": crypto/rsa: 512-bit keys are insecure (see https://go.dev/pkg/crypto/rsa#hdr-Minimum_key_size)
    sig0_test.go:33: failed to generate key for "RSASHA256": crypto/rsa: 512-bit keys are insecure (see https://go.dev/pkg/crypto/rsa#hdr-Minimum_key_size)
--- FAIL: TestTXTEscapeJustParse (0.00s)
    tester_test.go:37: example.com.     3600    IN      TXT     "unquoted"
    tester_test.go:37: example.com.     3600    IN      TXT     "quoted"
    tester_test.go:35: mismatch after parsing `"innerescaped\"quote"` TXT record: `innerescaped\"quote` != `innerescaped"quote`
    tester_test.go:37: example.com.     3600    IN      TXT     "innerescaped\"quote"
    tester_test.go:37: example.com.     3600    IN      TXT     "1backslash"
    tester_test.go:37: example.com.     3600    IN      TXT     "2back\\slash"
    tester_test.go:37: example.com.     3600    IN      TXT     "3back\\slash"
    tester_test.go:37: example.com.     3600    IN      TXT     "4back\\\\slash"
    tester_test.go:37: example.com.     3600    IN      TXT     "5back\\\\slash"
    tester_test.go:35: mismatch after parsing `"q1back\slash"` TXT record: `q1back\slash` != `1backslash`
    tester_test.go:37: example.com.     3600    IN      TXT     "q1backslash"
    tester_test.go:35: mismatch after parsing `"q2back\\slash"` TXT record: `q2back\\slash` != `2back\slash`
    tester_test.go:37: example.com.     3600    IN      TXT     "q2back\\slash"
    tester_test.go:35: mismatch after parsing `"q3back\\\slash"` TXT record: `q3back\\\slash` != `3back\slash`
    tester_test.go:37: example.com.     3600    IN      TXT     "q3back\\slash"
    tester_test.go:35: mismatch after parsing `"q4back\\\\slash"` TXT record: `q4back\\\\slash` != `4back\\slash`
    tester_test.go:37: example.com.     3600    IN      TXT     "q4back\\\\slash"
    tester_test.go:35: mismatch after parsing `"q5back\\\\\slash"` TXT record: `q5back\\\\\slash` != `5back\\slash`
    tester_test.go:37: example.com.     3600    IN      TXT     "q5back\\\\slash"
FAIL
FAIL    github.com/miekg/dns    1.002s
ok      github.com/miekg/dns/dnsutil    0.002s
FAIL
```

--- a/parse_test.go
+++ b/parse_test.go
@@ -1250,8 +1250,8 @@ func TestNewPrivateKey(t *testing.T) {
 	algorithms := []algorithm{
 		{ECDSAP256SHA256, 256},
 		{ECDSAP384SHA384, 384},
-		{RSASHA1, 512},
-		{RSASHA256, 512},
+		{RSASHA1, 1024},
+		{RSASHA256, 1024},
 		{ED25519, 256},
 	}
 

--- a/sig0_test.go
+++ b/sig0_test.go
@@ -25,7 +25,7 @@ func TestSIG0(t *testing.T) {
 			keysize = 256
 		case ECDSAP384SHA384:
 			keysize = 384
-		case RSASHA512:
+		case RSASHA1, RSASHA256, RSASHA512:
 			keysize = 1024
 		}
 		pk, err := keyrr.Generate(keysize)

